core/src/main/resources/error/error-classes.json

```diff
@@ -795,6 +795,11 @@
       }
     }
   },
+  "INVALID_SQL_ARG" : {
+    "message" : [
+      "The argument <name> of `sql()` is invalid. Consider to replace it by a SQL literal statement."
```

Comment:
Can we more explicit about why it is invalid?
"The argument <name> of `sql()` is not a literal." What is a "SQL Literal statement"?

Reply:
> What is a "SQL Literal statement"?

Any SQL statement that can produce a literal, see https://spark.apache.org/docs/latest/sql-ref-literals.html

Reply:
That's an expression then. Statements are top level. (Like SELECT, UPDATE, CRAETE, SET).

Reply:
how about just say `SQL literal`?

Reply:
I will change this in https://github.com/apache/spark/pull/39183